tests/MiddlewareTest.php

```diff
@@ -17,12 +18,48 @@ class MiddlewareTest extends TestCase
      */
     private $middleware;
 
-
     public function setUp()
     {
         $this->middleware = new Middleware;
     }
 
+    public function createHandler($handlerReturnValue): stdClass
+    {
+        $handler = new class extends stdClass
```

Comment:
This seems overly complex for the test requirements. It introduces _too_ much logic into the test that isn't tested. It would be better to keep it simple.  
What about if you keep the closures added to the middleware and `use` a local variable by reference. In each closure, you can modify/increment the variable and assert its value after executing the middleware. That's much simpler and should still achieve the desired result of improving tests

Reply:
Thanks for the feedback. I agree with you that this change was too complicated for the test. I have changed my approach as you have recommended. Please let me know what you think 😃